source/src/Internal.cc

```diff
@@ -62,9 +62,62 @@ namespace dqm4hep {
       /* WINDOWS implementation: TBD */
     }
 #elif __APPLE__
-    void memStats(MemoryStats &) {
-      /* MACOSX implementation: TBD */
+    void memStats(MemoryStats &stats) {
+      double unit = 1024. * 1024.; // Store everything in mb
+
+      // Swap info
+      // Swap is dynamically alotted on OSX, It will expand as the system needs it until the boot drive is filled.
+      struct xsw_usage swapStats;
+      size_t lengthSwap = sizeof(swapStats);
+      if (KERN_SUCCESS != sysctlbyname("vm.swapusage", &swapStats, &lengthSwap, NULL, 0))
+        std::cerr << "[OSX] - Failed to get swap statistics" << std::endl;
```

Comment:
Can you replace these `std::cerr` by `dqm_error("message")` and throw a `StatusCodeException` ?

Reply:
Yes, this was temporary and forgot to change it. Fixed in https://github.com/DQM4HEP/dqm4hep-core/pull/34/commits/22cf2aee7d2b138c40605e89cc81b406c796df6d